Classes/Aggregate/BackendPreviewAggregate.php

```diff
@@ -192,19 +179,38 @@ public function preProcess(PageLayoutView &\$parentObject, &\$drawItem, &\$heade
     }
 
     /**
-     * This function is needed for testing purpose
-     *
      * @param array \$pageTsConfig
      * @param string \$contentType
-     * @return string
+     * @return StandaloneView
      */
-    protected function getTemplatePath(array \$pageTsConfig, \$contentType)
+    protected function getViewFromConfiguration(array \$pageTsConfig, \$contentType)
     {
-        if (empty(\$pageTsConfig['mod.']['web_layout.']['tt_content.']['preview.'][\$contentType])) {
-            return '';
+        \$objectManager = GeneralUtility::makeInstance(ObjectManager::class);
+        \$view = \$objectManager->get(StandaloneView::class);
+        \$view->getRenderingContext()->setControllerName('Content');
+        \$view->getRenderingContext()->setControllerAction(\$this->supportedContentTypes[\$contentType]);
+
+        if (empty(\$pageTsConfig['mod.']['web_layout.']['tt_content.']['preview.'])) {
+            return \$view;
+        }
+
+        \$previewConfiguration = \$pageTsConfig['mod.']['web_layout.']['tt_content.']['preview.'];
+        \$extensionKey = substr(\$contentType, 0, -strlen(\$this->supportedContentTypes[\$contentType])) . '.';
+        if (!empty(\$previewConfiguration[\$extensionKey]['templateRootPath'])) {
+            \$view->setTemplateRootPaths([\$previewConfiguration[\$extensionKey]['templateRootPath']]);
+        }
+        if (!empty(\$previewConfiguration[\$extensionKey]['layoutRootPath'])) {
+            \$view->setLayoutRootPaths([\$previewConfiguration[\$extensionKey]['layoutRootPath']]);
+        }
+        if (!empty(\$previewConfiguration[\$extensionKey]['partialRootPath'])) {
+            \$view->setPartialRootPaths([\$previewConfiguration[\$extensionKey]['partialRootPath']]);
+        }
```

Comment:
Add `\$view->setTemplate(ucfirst(\$this->supportedContentTypes[\$contentType]));` here (before setting template path and file name